src/middleware/security.ts

```
@@ -0,0 +1,251 @@
+  next();
+};
+
+// Roll number format validator (standalone function)
+export const isValidRollNumber = (rollNumber: string): boolean => {
+  const rollRegex = /^\d{2}[a-zA-Z0-9]{2}[a-zA-Z0-9]{6}$/;
+  return rollRegex.test(rollNumber.toUpperCase().trim());
+};
+
+// Bot-specific security middleware
+export const createBotSecurityHandler = () => {
+  const userRequestCounts = new Map<number, { count: number; resetTime: number }>();
```

ESLint

[error] Replace `number,·{·count:·number;·resetTime:·number·}` with `⏎····number,⏎····{·count:·number;·resetTime:·number·}⏎··`

(prettier/prettier)